elfpy/data/postgres.py

```diff
@@ -527,34 +606,25 @@ def get_latest_block_number(session: Session) -> int:
     int
         The latest block number in the poolinfo table
     """
-
-    # query_results = session.query(PoolInfoTable).order_by(PoolInfoTable.timestamp.desc()).first()
-    query_results = session.query(PoolInfo).order_by(PoolInfo.timestamp.desc()).first()
-    # If the table is empty, query_results will return None
-    if query_results is None:
-        return 0
-    return int(query_results.blockNumber)
+    return get_latest_block_number_from_table("poolinfo", session)
 
 
-def _get_latest_block_number_wallet_info(session: Session) -> int:
-    """
-    Gets the latest block number based on the walletinfo table in the db
-    This function shouldn't be called externally, as the pool info table should be the main keeper of block numbers
-    """
-    query_results = session.query(WalletInfo).order_by(WalletInfo.id.desc()).first()
-    # If the table is empty, query_results will return None
-    if query_results is None:
-        return 0
-    return int(query_results.blockNumber)
+def get_latest_block_number_from_table(table: str, session: Session) -> int:
+    """Gets the latest block number based on the specified table in the db.
 
+    Arguments
+    ---------
+    table : str
+        The name of the table to get the latest block number from
+    session : Session
+        The initialized session object
 
-def _get_latest_block_number_transactions(session: Session) -> int:
-    """
-    Gets the latest block number based on the transactions table in the db
-    This function shouldn't be called externally, as the pool info table should be the main keeper of block numbers
+    Returns
+    -------
+    int
+        The latest block number from the specified table
     """
-    query_results = session.query(Transaction).order_by(Transaction.id.desc()).first()
-    # If the table is empty, query_results will return None
-    if query_results is None:
+    result = session.execute(text(f'SELECT MAX("{table}"."blockNumber") FROM {table}')).fetchone()
+    if result is None:
         return 0
-    return int(query_results.blockNumber)
+    return int(result[0] or 0)
```

Comment:
why do you need `or 0` if you have the `if result is None` conditional?

Reply:
that first step checks `result` which can itself be None, and can also contain `(None,)` so this is a second None check on that first element